MapboxNavigation/RouteMapViewController.swift

```diff
@@ -299,6 +303,43 @@ class RouteMapViewController: UIViewController, PulleyPrimaryContentControllerDe
             }
         }
     }
+
+    func addDebugAnnotation(location: CLLocation) {
+        let annotation = DebugAnnotation()
+        annotation.coordinate = location.coordinate
+        annotation.title = "Reroute"
+        annotation.color = .red
+        let metersInFrontOfUser = location.speed * RouteControllerDeadReckoningTimeInterval
+        let locationInfrontOfUser = location.coordinate.coordinate(at: metersInFrontOfUser, facing: location.course)
+        let newLocation = CLLocation(latitude: locationInfrontOfUser.latitude, longitude: locationInfrontOfUser.longitude)
+        let radius = min(RouteControllerMaximumDistanceBeforeRecalculating,
+                         location.horizontalAccuracy + RouteControllerUserLocationSnappingDistance)
```

Comment:
It would be safer to factor out and reuse this logic from here and `RouteController` as @1ec5 suggested.